### PR TITLE
Impl mkldnn batch norm

### PR DESCRIPTION
**Pull request by okdshin**
his PR depends on #131
Add MKLDNN BatchNromalization operator for mkldnn_with_generic_fallback backend

### CODE REVIEW

File: menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/batch_norm.hpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+                std::vector<mkldnn::primitive> primitives;
+                std::vector<std::pair<std::string, memory_cache>>
+                  temp_memory_cache_list;
+
+                auto epsilon = attribute_float(node, "epsilon");
+                auto spatial = attribute_int(node, "spatial");
+                if(!spatial) {
+                    throw std::runtime_error(
+                      "Non spacial BatchNorm is not supported");
```
Comment (msakai) [inline menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/batch_norm.hpp]: It might be better to use official operator name "BatchNormalization".

Comment (okdshin) [reply menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/batch_norm.hpp]: I see. I will fix.

File: menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/batch_norm.hpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    std::copy(
+                      static_cast<float*>(b_memory.get_data_handle()),
+                      static_cast<float*>(b_memory.get_data_handle()) +
+                        scale_total_size,
+                      static_cast<float*>(weight_memory.get_data_handle()) +
+                        scale_total_size);
+                } else {
+                    throw std::runtime_error(
+                      "BatchNormalization with invalid data_type");
+                }
```
Comment (msakai) [inline menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/batch_norm.hpp]: It seems that the copy from `scale_memory`/`b_memory` to `weight_memory` is performed in the MKL-DNN computation graph construction time, not execution time. Isn't this problematic when `scale_memory`/`b_memory`  are not constant tensor?

Comment (msakai) [reply menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/batch_norm.hpp]: I see.

**Comment by okdshin** [comment]
Thanks for review!
